File: 1. Fundamental-Level/CSharp/03. OOP-November-2015/[HW]/03. Exception-Handling/03. [EX]/01. Valid-Person/Person.cs

using System;

public class Person
{
    private string firstName;
    private string lastName;
    private int age;

    public Person(string firstName, string lastName, int age)
    {
        this.FirstName = firstName;
        this.LastName = lastName;
        this.Age = age;
    }

    public string FirstName
    {
        get
        {
            return this.firstName;
        }
        set
        {
            if (string.IsNullOrWhiteSpace(value))
            {
                throw new ArgumentNullException("The first name cannot be empty.");
            }

            this.firstName = value;
        }
    }

    public string LastName
    {
        get
        {
            return this.lastName;
        }
        set
        {
            if (string.IsNullOrWhiteSpace(value))
            {
                throw new ArgumentNullException("The last name cannot be empty.");
            }

            this.lastName = value;
        }
    }

    public int Age
    {
        get
        {
            return this.age;
        }
        set
        {
            if (value < 0 || value > 120)
            {
                throw new ArgumentOutOfRangeException("Age", "Invalid age.");
            }

            this.age = value;
        }
    }

    public override string ToString()
    {
        return string.Format("{0} {1}, Age: {2}", this.FirstName, this.LastName, this.Age);
    }
}

File: 1. Fundamental-Level/Java/OOP-Bacis-June-2016/[EX]/01. Defining_Classes/src/problem2_CreatingConstructors/Person.java
package problem2_CreatingConstructors;

public class Person {
   public String name;
   public int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public Person() {
        this("No name", 1);
    }

    public Person(int age) {
        this("No name", age);
    }

    public String getName() {
        return this.name;
    }

    public int getAge() {
        return this.age;
    }
}